File: Project 1/java/src/project1/Blog.java
package project1;

public class Blog {
    private int id;
    private String title;
    private String content;
    private String author;

    public Blog(int id, String title, String content, String author) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.author = author;
    }

    public Blog(String title, String content, String author) {
        this.title = title;
        this.content = content;
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean updateTitle(String newTitle) {
        if (newTitle != null && !newTitle.isEmpty()) {
            this.title = newTitle;
            return true;
        }
        return false;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public boolean updateContent(String newContent) {
        if (newContent != null && !newContent.isEmpty()) {
            this.content = newContent;
            return true;
        }
        return false;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "Blog{" + "id=" + id + ", title='" + title + "', content='" + content + "', author='" + author + "'}";
    }

}

File: Project 1/javascript/src/backend/models/Blog.js
class Blog {
  constructor(id, title, content, author) {
    this.id = id || 0;
    this.title = title || '';
    this.content = content || '';
    this.author = author || '';
    this.createdAt = new Date();
  }

  getTitle() {
    return this.title;
  }

  setTitle(title) {
    this.title = title;
  }

  updateTitle(newTitle) {
    if (newTitle && newTitle.trim() !== '') {
      this.title = newTitle;
      return true;
    }
    return false;
  }

  getContent() {
    return this.content;
  }

  setContent(content) {
    this.content = content;
  }

  updateContent(newContent) {
    if (newContent && newContent.trim() !== '') {
      this.content = newContent;
      return true;
    }
    return false;
  }

  getAuthor() {
    return this.author;
  }

  setAuthor(author) {
    this.author = author;
  }

  getId() {
    return this.id;
  }

  setId(id) {
    this.id = id;
  }

  getCreatedAt() {
    return this.createdAt;
  }

  toJSON() {
    return {
      id: this.id,
      title: this.title,
      content: this.content,
      author: this.author,
      createdAt: this.createdAt
    };
  }
}